### PR TITLE
Make Sampler a member of the class for llm/vlm pipelines

### PR DESCRIPTION
**Pull request by sbalandi**
cherry-pick https://github.com/openvinotoolkit/openvino.genai/pull/1347 to master

### CODE REVIEW

File: src/cpp/src/llm_pipeline.cpp [hunk 1]
```diff
@@ -96,6 +97,9 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
         // If eos_token_id was not provided, take value
         if (m_generation_config.eos_token_id == -1)
             m_generation_config.set_eos_token_id(m_tokenizer.get_eos_token_id());
+
+        m_sampler = Sampler(m_tokenizer);
```
Comment (Wovchena) [inline src/cpp/src/llm_pipeline.cpp:101]: You can move this to member initializer list for both pipelines

File: src/cpp/src/llm_pipeline.cpp [hunk 2]
```diff
@@ -358,9 +362,15 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
...
+            if (m_sampler.get_seed() != config.rng_seed) {
+                m_sampler.set_seed(config.rng_seed);
+            }
+
             std::tie(result, m_selected_beam) = ov::genai::get_lm_encoded_results(m_model_runner, input_ids, concatenated_attention_mask, streamer_ptr,
-                                                                                  sampler, requests, position_ids, std::nullopt, m_selected_beam);
+                                                                                  m_sampler, requests, position_ids, std::nullopt, m_selected_beam);
+
+            for (auto& request : requests)
+                m_sampler.clear_request_info(request->get_request_id());
```
Comment (Wovchena) [inline src/cpp/src/llm_pipeline.cpp:373]: You clear requests in both pipelines after `get_lm_encoded_results()`. Move it to `get_lm_encoded_results()`

File: src/cpp/src/visual_language/pipeline.cpp [hunk 3]
```diff
@@ -296,6 +307,8 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
         m_language.reset_state();
         // clear all chat history
         m_inputs_embedder->finish_chat();
+        // clear sampler requests info
+        m_sampler.clear_request_info(0);
```
Comment (Wovchena) [inline src/cpp/src/visual_language/pipeline.cpp:311]: It's already cleared after `get_lm_encoded_results()`. Why is it cleared again?

File: src/cpp/src/llm_pipeline.cpp (hunk 2 above)
Comment (sbalandi) [reply src/cpp/src/llm_pipeline.cpp]: moved

File: src/cpp/src/visual_language/pipeline.cpp (hunk 3 above)
Comment (sbalandi) [reply src/cpp/src/visual_language/pipeline.cpp]: removed

File: src/cpp/src/llm_pipeline.cpp (hunk 1 above)
Comment (sbalandi) [reply src/cpp/src/llm_pipeline.cpp:101]: I'm not sure I understand right what do you mean, do you mean moved it here https://github.com/openvinotoolkit/openvino.genai/blob/af3d3e12f5f1bc6b8c0b6720c93327313f630fa4/src/cpp/src/llm_pipeline.cpp#L48 ?

Comment (Wovchena) [reply src/cpp/src/llm_pipeline.cpp:101]: No. There's a special initialization form in C++ called `member initializer list`. It should look like
```cpp
    StatefulLLMPipeline(
        const std::shared_ptr<ov::Model>& model,
        const ov::genai::Tokenizer& tokenizer,
        const std::string& device,
        const ov::AnyMap& config,
        const ov::genai::GenerationConfig& generation_config
    ) : 
        LLMPipelineImplBase(tokenizer, generation_config),
        m_sampler(m_tokenizer) {
```

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**